example-code/snippets/src/pcf/DynamicSchema.ts

```diff
@@ -0,0 +1,205 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Schema as MetaSchema, SchemaContext, SchemaLoader, SchemaPropsGetter } from "@itwin/ecschema-metadata";
+import { ClassRegistry, Element, ElementAspect, IModelDb, Relationship, Schema, Schemas } from "@itwin/core-backend";
+import { AnyDiagnostic, ISchemaChanges, ISchemaCompareReporter, SchemaChanges, SchemaComparer, SchemaContextEditor } from "@itwin/ecschema-editing";
+import { MutableSchema } from "@itwin/ecschema-metadata/lib/cjs/Metadata/Schema";
+import * as pcf from "./pcf";
+
+export interface DynamicEntityMap {
+  entities: {
+    props: pcf.ECDynamicEntityClassProps;
+    registeredClass?: typeof Element | typeof ElementAspect;
+  }[];
+  relationships: {
+    props: pcf.ECDynamicRelationshipClassProps;
+    registeredClass?: typeof Relationship;
+  }[];
+}
+
+export interface DynamicSchemaProps {
+  schemaName: string;
+  schemaAlias: string;
+  dynamicEntityMap: DynamicEntityMap;
+}
+
+export interface SchemaVersion {
+  readVersion: number;
+  writeVersion: number;
+  minorVersion: number;
+}
+
+export async function tryGetSchema(db: IModelDb, schemaName: string): Promise<MetaSchema | undefined> {
+  const loader = new SchemaLoader((name) => db.getSchemaProps(name));
+  const schema = loader.tryGetSchema(schemaName);
+  return schema;
+}
+
+// PUBLISH_EXTRACT_START TestConnector-syncDynamicSchema.example-code
+export async function syncDynamicSchema(
+  db: IModelDb,
+  domainSchemaNames: string[],
+  props: DynamicSchemaProps
+): Promise<pcf.ItemState> {
+
+  const { schemaName } = props;
+  const existingSchema = await tryGetSchema(db, schemaName);
```

Comment:
Shouldn't the definition of this function be included in the published snippet?

Reply:
@pmconne Thanks for having a look.  I put a link to the complete source at the top of the article.  I was trying to find a balance between highlighting the main steps to creating a dynamic schema and pulling in more and more of the source.  I can add tryGetSchema if you think it will make the documentation more clear.

Reply:
No, that makes sense - thanks.